Modules/ECG-Baseline/ECG_Baseline.cpp

```diff
@@ -0,0 +1,203 @@
+
+    for (size_t currentIndex = 0; currentIndex < originalSignal->size(); currentIndex++)
+    {
+        sum = CalculateConvolutionResultForChosenElementOfSignal(originalSignal, m_LMSFilterParameters.weights, currentIndex);
+        error = m_LMSFilterParameters.desireSignal.at(currentIndex) - sum;
+        m_LMSFilterParameters.errors.push_back(error);
+        UpdateLMSWeights(error);
+        m_LMSFilterParameters.filteredSignal.push_back(sum);
+    }
+}
+
+void ECG_Baseline::CreateBWBandPassFilter()
+{
+    // Vector of filter coefficients
+    std::vector<float> vectorOfCoefficients;
+    int firstValue = -m_ButterworthFilterParameters.coefficientsNumber;
+    for (int i = 0; i < 2 * m_ButterworthFilterParameters.coefficientsNumber + 1; i++)
+    {
+        vectorOfCoefficients.push_back(firstValue);
+        firstValue++;
+    }
+
+    // Frequency normalization
+    unsigned int normfLowPass = m_ButterworthFilterParameters.fLowPass / (m_ButterworthFilterParameters.fSampling / 2);
+    unsigned int normfHighPass = m_ButterworthFilterParameters.fHighPass / (m_ButterworthFilterParameters.fSampling / 2);
+
+    // LowPass and HighPass Filter Coeff
+    for (auto& coeff : vectorOfCoefficients)
+    {
+        if (coeff == 0)
+        {
+            m_ButterworthFilterParameters.lowPassFilter.push_back(2 * normfLowPass);
+            m_ButterworthFilterParameters.highPassFilter.push_back(1 - 2 * normfHighPass);
+        }
+        else
+        {
+            m_ButterworthFilterParameters.lowPassFilter.push_back(sin((2 * M_PIl * normfLowPass * coeff) / (M_PIl * coeff)));
+            m_ButterworthFilterParameters.highPassFilter.push_back(-sin((2 * M_PIl * normfHighPass * coeff) / (M_PIl * coeff)));
+        }
+    }
+
+    // Create HammingWindow
+    for (size_t i = 0; i < vectorOfCoefficients.size(); i++)
+    {
+        m_ButterworthFilterParameters.hammingWindow.push_back(0.54 - 0.46 * cos((2 * M_PIl * i) / (vectorOfCoefficients.size())));
+    }
+
+    // Windowing 
+    for (size_t i = 0; i < vectorOfCoefficients.size(); i++)
+    {
+        m_ButterworthFilterParameters.lowPassFilter.at(i) = m_ButterworthFilterParameters.lowPassFilter.at(i) * m_ButterworthFilterParameters.hammingWindow.at(i);
+        m_ButterworthFilterParameters.highPassFilter.at(i) = m_ButterworthFilterParameters.highPassFilter.at(i) * m_ButterworthFilterParameters.hammingWindow.at(i);
+    }
+
+}
+void ECG_Baseline::UpdateLMSWeights(float error)
+{
+    for (size_t i = 0; i < m_LMSFilterParameters.currentlyCalculatedSubvector.size(); i++)
+    {
+        m_LMSFilterParameters.weights.at(m_LMSFilterParameters.weights.size() - 1 - i) = m_LMSFilterParameters.weights.at(m_LMSFilterParameters.weights.size() - 1 - i) + (error * m_LMSFilterParameters.currentlyCalculatedSubvector.at(i) * m_LMSFilterParameters.convergenceRate);
+    }
+}
+
+void ECG_Baseline::CreateMAFConvolutionWindow()
+{
+    float maskValue = (float)1. / m_MovingAverageFilterParameters.windowLenght;
```

Comment:
Co jeżeli `m_MovingAverageFilterParameters.windowLenght` przyjdzie jako 0?

Reply:
to bedzie sprawdozne na ViewModelu przed wywolaniem metod filtracji